services/ui-src/src/components/fields/DynamicField.tsx

```diff
@@ -141,6 +141,19 @@ export const DynamicField = ({ name, label, ...props }: Props) => {
         }
       );
 
+      // filter analysis methods to remove deleted plans
+      const filteredAnalysisMethods = report?.fieldData?.analysisMethods?.map(
+        (method: EntityShape) => {
+          if (method.analysis_method_applicable_plans?.length) {
+            method.analysis_method_applicable_plans =
+              method.analysis_method_applicable_plans.filter(
+                (plan: AnyObject) => plan.key !== selectedRecord.id
+              );
+          }
+          return method;
+        }
+      );
```

Comment:
Let me know if you can think of a cleaner way to make this edit! I got this from combining patterns above here in the file. I think it's tricky because there is double nesting of arrays and objects to get to what we need to edit. I played around with a for...of but still needed some kind of map/filter loop to do the changes.

Reply:
I think this could be an opportunity for a refactor in the future, but it looks good to me for now 💯